io/net/net_tools.py

```diff
@@ -0,0 +1,289 @@
+    """
+    Net_tools class verify the functionality of various network tools.
+    hostname: set/get hostname by cmd. Check it with the expected result.
+    ifconfig: setup an alias for lo. Check it can be listed.
+    netstat: run it with different options. Check the return code.
+    arp: run it with different options. Check the return code.
+    traceroute,traceroute6: run for localhost. It should report 1 hop.
+    route: run it with -n. Check the return code.
+    ipmaddr: run it and check the return code.
+    iptunnel: create sit1 and check it can be list. Then remove it and
+              check it is removed from the list.
+    """
+
+    def setUp(self):
+        """
+        Verify Required tools are installed or not.
+        """
+        if os.getuid() != 0:
+            exit("You need to have root privileges to execute this script. \
+                 \nPlease try again, using 'sudo'. Exiting.")
+        sm = SoftwareManager()
+        self.detected_distro = distro.detect()
+        if self.detected_distro.name == "SuSE":
+            net_tools = ("net-tools", "traceroute")
+        else:
+            net_tools = ("net-tools", "hostname", "traceroute")
+        for pkg in net_tools:
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.error("%s package is need to test" % pkg)
+        self.ipv6 = False
+        if os.path.exists("/proc/net/if_inet6"):
+            self.ipv6 = True
+        interface_out = process.system_output("ip route show default",
+                                              verbose=False,
+                                              shell=True)
+        search_obj = re.search(
+            "default via\s+(\S+)\s+dev\s+(\w+)", interface_out)
+        active_iface = search_obj.group(2)
+        self.default_router = search_obj.group(1)
+        if active_iface is None:
+            self.error("can't find active network interface")
+        tmpdir = data_dir.get_tmp_dir()
+        self.backup_dir = os.path.join(tmpdir, 'net-tools')
+        if not os.path.isdir(self.backup_dir):
+            os.makedirs(self.backup_dir)
+        # Get Hostname
+        output = process.run(cmd="hostname", shell=True, verbose=False)
+        if output.exit_status != 0:
+            self.error("Failed To get hostname\n")
+        else:
+            if output.stdout is None:
+                # set hostname if not set
+                if not process.system("hostname localhost.localdomain"):
+                    self.error("Failed to set hostname\n")
+            process.system("hostname > %s/HOSTNAME" % self.backup_dir,
+                           verbose=False,
+                           shell=True)
+
+    def test_hostname(self):
+        """
+        Verify the functionality of the hostname command
+        """
+        hostname_cmd = "hostname"
+        output = process.run(cmd=hostname_cmd, shell=True, verbose=False)
+        if output.exit_status != 0 and output.stdout is None:
```

Comment:
Although the `process.run` raises exception in case the exit_status is non-zero, so you don't need to check for it (or you need to add `ignore_status=True` as `process.run` argument.